src/components/cesium_map/cesium_UI.js

```diff
@@ -170,14 +185,31 @@ class CesiumMap extends React.Component {
   // https://medium.com/@garrettmac/reactjs-how-to-safely-manipulate-the-dom-when-reactjs-cant-the-right-way-8a20928e8a6
   // manipulate Dom outside the react model
   shouldComponentUpdate(nextProps) {
-    const key1 = JSON.stringify(nextProps.searchFields)
-    const key2 = JSON.stringify(this.props.searchFields)
-    if (key1 !== key2) {
+    // update point primitive based on searchFields
+    const sf1 = JSON.stringify(nextProps.searchFields)
+    const sf2 = JSON.stringify(this.props.searchFields)
+    if (sf1 !== sf2) {
       // clear all element in cesium
       searchFields = nextProps.searchFields;
       updatePrimitive(viewer.currentView.latitude, viewer.currentView.longitude);
     }
 
+    // update bounding box based on bbox
+    const bb1 = JSON.stringify(nextProps.bbox);
+    const bb2 = JSON.stringify(this.props.bbox);
+
+    if (bb1 !== bb2 && bb1 !== JSON.stringify(bboxLoc)) {
```

Comment:
Prevent the app to rerender the map if the users draw a bounding box in the map.